lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1363,19 +1363,6 @@ propagateTensorOperand(SILValue v,
           return lastRootValue; // Cannot handle this.
         continue;
       }
-
-      // Otherwise simplify inputs in predecessor blocks.
-      for (auto pi : arg->getParent()->getPredecessorBlocks()) {
```

Comment:
Is this no longer needed because we look at all the operands now? If that is the case, shouldn't the iteration below ensure all its predecessors are processed before a node is processed?

Reply:
Reverted this change.